File: Common/Conversion.vb
Imports System.Runtime.CompilerServices

Public NotInheritable Class ParseInv

    Private Sub New()
        Throw New NotSupportedException()
    End Sub

    Public Shared Function [Integer](ByVal Str As String) As Integer
        Return Integer.Parse(Str, Globalization.CultureInfo.InvariantCulture)
    End Function

    Public Shared Function [Long](ByVal Str As String) As Long
        Return Long.Parse(Str, Globalization.CultureInfo.InvariantCulture)
    End Function

    Public Shared Function [Short](ByVal Str As String) As Short
        Return Short.Parse(Str, Globalization.CultureInfo.InvariantCulture)
    End Function

    Public Shared Function [Byte](ByVal Str As String) As Byte
        Return Byte.Parse(Str, Globalization.CultureInfo.InvariantCulture)
    End Function

    Public Shared Function [UInteger](ByVal Str As String) As UInteger
        Return UInteger.Parse(Str, Globalization.CultureInfo.InvariantCulture)
    End Function

    Public Shared Function [ULong](ByVal Str As String) As ULong
        Return ULong.Parse(Str, Globalization.CultureInfo.InvariantCulture)
    End Function

    Public Shared Function [UShort](ByVal Str As String) As UShort
        Return UShort.Parse(Str, Globalization.CultureInfo.InvariantCulture)
    End Function

    Public Shared Function [SByte](ByVal Str As String) As SByte
        Return SByte.Parse(Str, Globalization.CultureInfo.InvariantCulture)
    End Function

    Public Shared Function [Double](ByVal Str As String) As Double
        Return Double.Parse(Str, Globalization.CultureInfo.InvariantCulture)
    End Function

    Public Shared Function [Single](ByVal Str As String) As Single
        Return Single.Parse(Str, Globalization.CultureInfo.InvariantCulture)
    End Function

    Public Shared Function [Decimal](ByVal Str As String) As Decimal
        Return Decimal.Parse(Str, Globalization.CultureInfo.InvariantCulture)
    End Function

    Public Shared Function [Boolean](ByVal Str As String) As Boolean
        Return Boolean.Parse(Str)
    End Function

    Public Shared Function TryInteger(ByVal Str As String, ByRef Value As Integer) As Boolean
        Return Integer.TryParse(Str, Globalization.NumberStyles.Integer, Globalization.CultureInfo.InvariantCulture, Value)
    End Function

    Public Shared Function TryLong(ByVal Str As String, ByRef Value As Long) As Boolean
        Return Long.TryParse(Str, Globalization.NumberStyles.Integer, Globalization.CultureInfo.InvariantCulture, Value)
    End Function

    Public Shared Function TryShort(ByVal Str As String, ByRef Value As Short) As Boolean
        Return Short.TryParse(Str, Globalization.NumberStyles.Integer, Globalization.CultureInfo.InvariantCulture, Value)
    End Function

    Public Shared Function TryByte(ByVal Str As String, ByRef Value As Byte) As Boolean
        Return Byte.TryParse(Str, Globalization.NumberStyles.Integer, Globalization.CultureInfo.InvariantCulture, Value)
    End Function

    Public Shared Function TryUInteger(ByVal Str As String, ByRef Value As UInteger) As Boolean
        Return UInteger.TryParse(Str, Globalization.NumberStyles.Integer, Globalization.CultureInfo.InvariantCulture, Value)
    End Function

    Public Shared Function TryULong(ByVal Str As String, ByRef Value As ULong) As Boolean
        Return ULong.TryParse(Str, Globalization.NumberStyles.Integer, Globalization.CultureInfo.InvariantCulture, Value)
    End Function

    Public Shared Function TryUShort(ByVal Str As String, ByRef Value As UShort) As Boolean
        Return UShort.TryParse(Str, Globalization.NumberStyles.Integer, Globalization.CultureInfo.InvariantCulture, Value)
    End Function

    Public Shared Function TrySByte(ByVal Str As String, ByRef Value As SByte) As Boolean
        Return SByte.TryParse(Str, Globalization.NumberStyles.Integer, Globalization.CultureInfo.InvariantCulture, Value)
    End Function

    Public Shared Function TryDouble(ByVal Str As String, ByRef Value As Double) As Boolean
        Return Double.TryParse(Str, Globalization.NumberStyles.Float Or Globalization.NumberStyles.AllowThousands, Globalization.CultureInfo.InvariantCulture, Value)
    End Function

    Public Shared Function TrySingle(ByVal Str As String, ByRef Value As Single) As Boolean
        Return Single.TryParse(Str, Globalization.NumberStyles.Float Or Globalization.NumberStyles.AllowThousands, Globalization.CultureInfo.InvariantCulture, Value)
    End Function

    Public Shared Function TryDecimal(ByVal Str As String, ByRef Value As Decimal) As Boolean
        Return Decimal.TryParse(Str, Globalization.NumberStyles.Number, Globalization.CultureInfo.InvariantCulture, Value)
    End Function

    Public Shared Function TryBoolean(ByVal Str As String, ByRef Value As Boolean) As Boolean
        Return Boolean.TryParse(Str, Value)
    End Function

End Class

Public Module ConversionExtensions

    <Extension()>
    Public Function ToStringInv(ByVal Self As Integer) As String
        Return Self.ToString(Globalization.CultureInfo.InvariantCulture)
    End Function

    <Extension()>
    Public Function ToStringInv(ByVal Self As Long) As String
        Return Self.ToString(Globalization.CultureInfo.InvariantCulture)
    End Function

    <Extension()>
    Public Function ToStringInv(ByVal Self As Short) As String
        Return Self.ToString(Globalization.CultureInfo.InvariantCulture)
    End Function

    <Extension()>
    Public Function ToStringInv(ByVal Self As Byte) As String
        Return Self.ToString(Globalization.CultureInfo.InvariantCulture)
    End Function

    <Extension()>
    Public Function ToStringInv(ByVal Self As UInteger) As String
        Return Self.ToString(Globalization.CultureInfo.InvariantCulture)
    End Function

    <Extension()>
    Public Function ToStringInv(ByVal Self As ULong) As String
        Return Self.ToString(Globalization.CultureInfo.InvariantCulture)
    End Function

    <Extension()>
    Public Function ToStringInv(ByVal Self As UShort) As String
        Return Self.ToString(Globalization.CultureInfo.InvariantCulture)
    End Function

    <Extension()>
    Public Function ToStringInv(ByVal Self As SByte) As String
        Return Self.ToString(Globalization.CultureInfo.InvariantCulture)
    End Function

    <Extension()>
    Public Function ToStringInv(ByVal Self As Double) As String
        Return Self.ToString(Globalization.CultureInfo.InvariantCulture)
    End Function

    <Extension()>
    Public Function ToStringInv(ByVal Self As Single) As String
        Return Self.ToString(Globalization.CultureInfo.InvariantCulture)
    End Function

    <Extension()>
    Public Function ToStringInv(ByVal Self As Decimal) As String
        Return Self.ToString(Globalization.CultureInfo.InvariantCulture)
    End Function

    <Extension()>
    Public Function ToStringInv(ByVal Self As Boolean) As String
        Return Self.ToString()
    End Function

End Module

File: Common/Conversion.cs
using System;

namespace Ks.Common
{
        public static class ParseInv
        {

            public static int Integer(string Str)
            {
                return int.Parse(Str, System.Globalization.CultureInfo.InvariantCulture);
            }

            public static long Long(string Str)
            {
                return long.Parse(Str, System.Globalization.CultureInfo.InvariantCulture);
            }

            public static short Short(string Str)
            {
                return short.Parse(Str, System.Globalization.CultureInfo.InvariantCulture);
            }

            public static byte Byte(string Str)
            {
                return byte.Parse(Str, System.Globalization.CultureInfo.InvariantCulture);
            }

            public static uint UInteger(string Str)
            {
                return uint.Parse(Str, System.Globalization.CultureInfo.InvariantCulture);
            }

            public static ulong ULong(string Str)
            {
                return ulong.Parse(Str, System.Globalization.CultureInfo.InvariantCulture);
            }

            public static ushort UShort(string Str)
            {
                return ushort.Parse(Str, System.Globalization.CultureInfo.InvariantCulture);
            }

            public static sbyte SByte(string Str)
            {
                return sbyte.Parse(Str, System.Globalization.CultureInfo.InvariantCulture);
            }

            public static double Double(string Str)
            {
                return double.Parse(Str, System.Globalization.CultureInfo.InvariantCulture);
            }

            public static float Single(string Str)
            {
                return float.Parse(Str, System.Globalization.CultureInfo.InvariantCulture);
            }

            public static decimal Decimal(string Str)
            {
                return decimal.Parse(Str, System.Globalization.CultureInfo.InvariantCulture);
            }

            public static bool Boolean(string Str)
            {
                return bool.Parse(Str);
            }

            public static bool TryInteger(string Str, out int Value)
            {
                return int.TryParse(Str, System.Globalization.NumberStyles.Integer, System.Globalization.CultureInfo.InvariantCulture, out Value);
            }

            public static bool TryLong(string Str, out long Value)
            {
                return long.TryParse(Str, System.Globalization.NumberStyles.Integer, System.Globalization.CultureInfo.InvariantCulture, out Value);
            }

            public static bool TryShort(string Str, out short Value)
            {
                return short.TryParse(Str, System.Globalization.NumberStyles.Integer, System.Globalization.CultureInfo.InvariantCulture, out Value);
            }

            public static bool TryByte(string Str, out byte Value)
            {
                return byte.TryParse(Str, System.Globalization.NumberStyles.Integer, System.Globalization.CultureInfo.InvariantCulture, out Value);
            }

            public static bool TryUInteger(string Str, out uint Value)
            {
                return uint.TryParse(Str, System.Globalization.NumberStyles.Integer, System.Globalization.CultureInfo.InvariantCulture, out Value);
            }

            public static bool TryULong(string Str, out ulong Value)
            {
                return ulong.TryParse(Str, System.Globalization.NumberStyles.Integer, System.Globalization.CultureInfo.InvariantCulture, out Value);
            }

            public static bool TryUShort(string Str, out ushort Value)
            {
                return ushort.TryParse(Str, System.Globalization.NumberStyles.Integer, System.Globalization.CultureInfo.InvariantCulture, out Value);
            }

            public static bool TrySByte(string Str, out sbyte Value)
            {
                return sbyte.TryParse(Str, System.Globalization.NumberStyles.Integer, System.Globalization.CultureInfo.InvariantCulture, out Value);
            }

            public static bool TryDouble(string Str, out double Value)
            {
                return double.TryParse(Str, System.Globalization.NumberStyles.Float | System.Globalization.NumberStyles.AllowThousands, System.Globalization.CultureInfo.InvariantCulture, out Value);
            }

            public static bool TrySingle(string Str, out float Value)
            {
                return float.TryParse(Str, System.Globalization.NumberStyles.Float | System.Globalization.NumberStyles.AllowThousands, System.Globalization.CultureInfo.InvariantCulture, out Value);
            }

            public static bool TryDecimal(string Str, out decimal Value)
            {
                return decimal.TryParse(Str, System.Globalization.NumberStyles.Number, System.Globalization.CultureInfo.InvariantCulture, out Value);
            }

            public static bool TryBoolean(string Str, out bool Value)
            {
                return bool.TryParse(Str, out Value);
            }
        }

        public static class ConversionExtensions
        {
            public static string ToStringInv(this int Self)
            {
                return Self.ToString(System.Globalization.CultureInfo.InvariantCulture);
            }

            public static string ToStringInv(this long Self)
            {
                return Self.ToString(System.Globalization.CultureInfo.InvariantCulture);
            }

            public static string ToStringInv(this short Self)
            {
                return Self.ToString(System.Globalization.CultureInfo.InvariantCulture);
            }

            public static string ToStringInv(this byte Self)
            {
                return Self.ToString(System.Globalization.CultureInfo.InvariantCulture);
            }

            public static string ToStringInv(this uint Self)
            {
                return Self.ToString(System.Globalization.CultureInfo.InvariantCulture);
            }

            public static string ToStringInv(this ulong Self)
            {
                return Self.ToString(System.Globalization.CultureInfo.InvariantCulture);
            }

            public static string ToStringInv(this ushort Self)
            {
                return Self.ToString(System.Globalization.CultureInfo.InvariantCulture);
            }

            public static string ToStringInv(this sbyte Self)
            {
                return Self.ToString(System.Globalization.CultureInfo.InvariantCulture);
            }

            public static string ToStringInv(this double Self)
            {
                return Self.ToString(System.Globalization.CultureInfo.InvariantCulture);
            }

            public static string ToStringInv(this float Self)
            {
                return Self.ToString(System.Globalization.CultureInfo.InvariantCulture);
            }

            public static string ToStringInv(this decimal Self)
            {
                return Self.ToString(System.Globalization.CultureInfo.InvariantCulture);
            }

            public static string ToStringInv(this bool Self)
            {
                return Self.ToString();
            }
        }
    }
